tests/net/socket/socketpair/src/main.c

```diff
@@ -31,63 +21,75 @@ extern void test_socketpair_unsupported_calls(void);
 /* in fcntl.c */
 extern void test_socketpair_fcntl(void);
 
+/* in nonblock.c */
+extern void test_socketpair_read_nonblock(void);
+extern void test_socketpair_write_nonblock(void);
+
+/* in block.c */
+extern void test_socketpair_read_block(void);
+extern void test_socketpair_write_block(void);
+
+/* in closed_ends.c */
+extern void test_socketpair_close_one_end_and_read_from_the_other(void);
+extern void test_socketpair_close_one_end_and_write_to_the_other(void);
+
 /* in poll.c */
 extern void test_socketpair_poll_timeout(void);
 extern void test_socketpair_poll_timeout_nonblocking(void);
-extern void test_socketpair_poll_close_remote_end_POLLIN(void);
-extern void test_socketpair_poll_close_remote_end_POLLOUT(void);
 extern void test_socketpair_poll_immediate_data(void);
 extern void test_socketpair_poll_delayed_data(void);
+extern void test_socketpair_poll_close_remote_end_POLLIN(void);
+extern void test_socketpair_poll_close_remote_end_POLLOUT(void);
+
+/* work queue for tests that need an async event */
+static K_THREAD_STACK_DEFINE(test_socketpair_work_q_stack, 512);
+struct k_work_q test_socketpair_work_q;
 
 void test_main(void)
 {
 	k_thread_system_pool_assign(k_current_get());
 
+	k_thread_access_grant(k_current_get(), &test_socketpair_work_q.thread,
+	    &test_socketpair_work_q.queue,
+	    &test_socketpair_work_q_stack);
```

Comment:
I don't think you need to grant permission on the stack object in order for user threads to submit to the queue. The workqueue itself is being set up in supervisor mode (which requires no permissions), and the workqueue runs in supervisor mode since it is started with `k_work_q_start`. So if the only interaction with a user thread is to enqueue items the workqueue, I think permission on `&test_socketpair_work_q.queue` may be sufficient for descendant user threads.

Reply:
I was following the example in [tests/kernel/workq](https://github.com/zephyrproject-rtos/zephyr/blob/master/tests/kernel/workq/work_queue_api/src/main.c#L867). I could try without though.

Reply:
Yep, you were right ;-)